src/default_allocator.cpp

```diff
@@ -14,20 +14,45 @@
 
 #include "default_allocator.h"
 
+#include <fmt/format.h>
+
 namespace vsag {
 void*
 DefaultAllocator::Allocate(size_t size) {
-    return malloc(size);
+    auto ptr = malloc(size);
+    std::lock_guard<std::mutex> guard(set_mutex_);
+    allocated_ptrs_.insert(ptr);
```

Comment:
Debug code ?

Reply:
The code is using to ensure that the memory allocated by this allocator will be released by this allocator, while the memory not allocated by this allocator will not be released by this allocator.